benchmarks/attention/benchmark_attention.py

```diff
@@ -0,0 +1,193 @@
+    df.to_csv('times.csv',index=False)
+    torch.cuda.cudart().cudaProfilerStop()
+
+def parse_results(per_cudnn, per_flash, model):
+    filename = f'prof_{model}_cuda_gpu_trace.csv'
+    df = pd.read_csv(os.path.join('./',filename))
+    df_times = pd.read_csv('times.csv')
+    row = len(df_times.index)-1
+    
+    if per_cudnn > 0:
+        t_cudnn_all = df[df['Name'].str.contains('cudnn')]['Duration (ns)'].to_numpy()
+        t_cudnn_all = t_cudnn_all.reshape(-1, per_cudnn)
+        t_cudnn_avg = np.average(t_cudnn_all, axis=0)
+        df_times.loc[row, 'FusedAttention Kernels (fwd)'] = t_cudnn_avg[0]/1e6
+        df_times.loc[row, 'FusedAttention Kernels (bwd)'] = t_cudnn_avg[1:4].sum()/1e6
+        df_times.loc[row, 'FusedAttention Kernels (fwd+bwd)'] = t_cudnn_avg.sum()/1e6
+
+    if per_flash > 0:
+        t_flash_all = df[df['Name'].str.contains('void flash')]['Duration (ns)'].to_numpy()
+        t_flash_all = t_flash_all.reshape(-1, per_flash)
+        t_flash_avg = np.average(t_flash_all, axis=0)
+        df_times.loc[row, 'FlashAttention Kernels (fwd)'] = t_flash_avg[0]/1e6
+        df_times.loc[row, 'FlashAttention Kernels (bwd)'] = t_flash_avg[1:4].sum()/1e6
+        df_times.loc[row, 'FlashAttention Kernels (fwd+bwd)'] = t_flash_avg.sum()/1e6
+
+    if per_cudnn > 0 and per_flash > 0:
+        df_times.loc[row, 'Fused vs Flash Kernels Speedup (fwd+bwd)'] = \
+                df_times.loc[row, 'FlashAttention Kernels (fwd+bwd)'] / \
+                df_times.loc[row, 'FusedAttention Kernels (fwd+bwd)']
+    df_times.to_csv('times.csv',index=False)
+
+def main():
+    times = pd.DataFrame(
+            columns=[
+                'FusedAttention Module',
+                'FusedAttention Kernels (fwd)',
+                'FusedAttention Kernels (bwd)',
+                'FusedAttention Kernels (fwd+bwd)',
+                'FlashAttention Module',
+                'FlashAttention Kernels (fwd)',
+                'FlashAttention Kernels (bwd)',
+                'FlashAttention Kernels (fwd+bwd)',
+                'Fused vs Flash Kernels Speedup (fwd+bwd)',
+                ])
+    times.to_csv('times.csv',index=False)
+
+    for model in model_configs.keys():
+        config = model_configs[model]
+        fused_attn_supported, fused_attn_backend = _is_fused_attention_supported(
+            config, dtype, qkv_layout=qkv_layout,
+        )
+        fused_attn_supported = fused_attn_supported and not swa
+        flash_attn_supported = _is_flash_attention_supported(config)
+
+        prof_cmd = f"""nsys profile \
+                --capture-range=cudaProfilerApi \
+                --capture-range-end=stop-shutdown \
+                --force-overwrite=true \
+                --output=prof_{model} \
+                python -c "import benchmark_attention; \
+                benchmark_attention.benchmark_dot_product_attention(\
+                '{model}', {fused_attn_supported}, {flash_attn_supported})" """
+        os.system(prof_cmd)
```

Comment:
`os.system` is not safe. `subprocess` module is preferred than `os.system`. See https://docs.python.org/3/library/os.html#os.system